src/visualization/utilities_p4est_t8code.jl

```diff
@@ -5,6 +5,303 @@
 @muladd begin
 #! format: noindent
 
+#############################################################################
+# 2D
+
+# The P4estMesh and T8codeMesh can make use of the efficient search functionality
+# based on the tree structure to speed up the process of finding the
+# elements in which the curve points are located.
+# We hard-code `Float64` here since some parts will be accessed from a Julia
+# function called as callback from C. Thus, we cannot use multiple dispatch
+# easily and use concrete types instead. See the comment on
+# `SearchPointsInP4estT8codeMesh2DHelper` below.
+# TODO: Implement the efficient search functionality also for the 2D T8codeMesh.
+function unstructured_2d_to_1d_curve(u, mesh::P4estMesh{2, 2, Float64},
+                                     equations, dg::DGSEM, cache,
+                                     curve, slice,
+                                     point, nvisnodes,
+                                     solution_variables)
+    # TODO: Currently, the efficient search functionality is only implemented
+    #       for linear meshes. If the mesh is not linear, we fall back to the naive
+    #       but general approach.
+    if length(mesh.nodes) != 2
+        return unstructured_2d_to_1d_curve_general(u, mesh, equations,
+                                                   dg, cache,
+                                                   curve, slice,
+                                                   point, nvisnodes,
+                                                   solution_variables)
+    end
+    # From here on, we know that we only have to deal with a linear mesh.
+
+    # If no curve is defined, create an axis curve.
+    original_nodes = cache.elements.node_coordinates
+    if curve === nothing
+        curve = axis_curve(view(original_nodes, 1, :, :, :),
+                           view(original_nodes, 2, :, :, :),
+                           slice, point, nvisnodes)
+    end
```

Comment:
I see that in 3D we have this block directly defined in `PlotData1D` before we pass the `curve` to `unstructured_3d_to_1d_curve`. In 2D we have this here and a similar block also in  `unstructured_2d_to_1d_curve_general`. Would it make sense to follow the same approach as in 3D and move this part [here](https://github.com/trixi-framework/Trixi.jl/pull/2340/files#diff-d87df04f4d61c4e804e3230715504dcfd390b14716474a439361e3adbda97e10R673), such that we also don't need https://github.com/trixi-framework/Trixi.jl/pull/2340/files#diff-faf85b716f2bcf8f583bd014dcec74fa4a5f5f35ee8b644a64c0ef7c0a642b5dR885-R887 anymore or is there a good reason for this discrepancy?